### PR TITLE
Added functionality to write running server URIs to file on startup.

### PR DESCRIPTION
**Pull request by mark-terry**
Added functionality to write the running server URIs to file on startup, consistent with similar behaviour in Besu.

Signed-off-by: Mark Terry <mark.terry@consensys.net>

### CODE REVIEW

File: server/server-utils/src/main/java/com/quorum/tessera/server/utils/ServerURIFileWriter.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    serverList.forEach(
+        tesseraServer -> {
+          uriList.add(
+              String.format(
+                  "%s=%s",
+                  tesseraServer.getAppType().toString(), tesseraServer.getUri().toString()));
+        });
+
+    final File file = new File(dirPath.toFile(), fileName);
+    file.deleteOnExit();
```
Comment (macfarla) [inline server/server-utils/src/main/java/com/quorum/tessera/server/utils/ServerURIFileWriter.java:41]: this is what ensures the file gets deleted when Tessera shuts down, right?

File: server/server-utils/src/main/java/com/quorum/tessera/server/utils/ServerURIFileWriter.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    try {
+      uriList.add(
+          0,
+          String.format(
+              "#%s. This file will be deleted after the enclave is shutdown.", fileHeader));
+      Files.write(
+          Path.of(dirPath.toAbsolutePath() + "/" + fileName), uriList, Charset.defaultCharset());
+    } catch (final Exception e) {
+      LOGGER.warn(String.format("Error writing %s file", fileName), e);
```
Comment (macfarla) [inline server/server-utils/src/main/java/com/quorum/tessera/server/utils/ServerURIFileWriter.java]: should this be debug level if all other associated logging is debug level?

**Review by macfarla — DISMISSED** [review]
LGTM

**Comment by Krish1979** [comment]
This PR is no different to the original PR but without the CLI option. I still don't see the need for this feature.

**Comment by antonydenyer** [comment]
This feels like a rehash of https://github.com/ConsenSys/tessera/pull/1356

File: server/server-utils/src/main/java/com/quorum/tessera/server/utils/ServerURIFileWriter.java (hunk 1 above)
Comment (mark-terry) [reply server/server-utils/src/main/java/com/quorum/tessera/server/utils/ServerURIFileWriter.java:41]: Correct.

**Comment by macfarla** [comment]
closing in favour of #1445

**Pull request closed without merging**